router/router_std.go

```diff
@@ -27,24 +26,43 @@ func DefaultNotFoundHandler(w http.ResponseWriter, req *http.Request, _ Params)
 
 // ServerHTTP implements http.Handler
 func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	if !r.opts.NoCatchPanics {
+	if !r.opts.NoCatchPanics && r.PanicHandler != nil {
 		defer func() {
-			if v := recover(); v != nil && r.PanicHandler != nil {
+			if v := recover(); v != nil {
 				r.PanicHandler(w, req, v)
 			}
 		}()
 	}
 
-	u := req.URL.EscapedPath()
+	u, method := req.URL.Path, req.Method
```

Comment:
Isnt it safer still doing req.URL.EscapedPath()?

Reply:
We're matching in a map so is safe, and it's a lot faster that way and it's not needed.